### PR TITLE
feat: #99088 React 18 upgrade - Change echo core state management tool

### PR DESCRIPTION
**Pull request by CsabaTheKiss**
⚠️ This PR will not be merged directly to main branch ⚠️ 
After the review it will be merged into the react 18 upgrade branch.

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

-   [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
-   [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] I have updated the documentation accordingly
-   [x] I have added tests to cover my changes

### Description

We're planning to upgrade Echo to use React 18.
Issue:
The `@dbeining/react-atom` state management tool used by EchoCore is no longer maintained, and it does not support React 18.

Using a different, more up to date state manager tool: `zustand`.

Luckly the different state handler functions are atomic, well defined functions.
As a result only the underlying react-atom APIs had to be changed => no breaking changes!

### Remarks

React 18 upgrade branch for echoCore:
https://github.com/equinor/EchoCore/tree/patch/upgrade-to-react-18

Related PRs
* EchopediaWeb: https://github.com/equinor/EchopediaWeb/pull/1348
* EchoComponents: https://github.com/equinor/EchoComponents/pull/83
* EchoFramework: https://github.com/equinor/EchoFramework/pull/286
* EchoUtils: https://github.com/equinor/EchoUtils/pull/72

### CODE REVIEW

File: packages/echo-core/src/state/useGlobalState.ts [hunk 1]
```diff
@@ -16,6 +15,6 @@ export function useGlobalState(): GlobalState;
 export function useGlobalState<R>(select: (state: GlobalState) => R): R;
 
 export function useGlobalState<R>(select?: (state: GlobalState) => R): GlobalState | R {
```
Comment (OveBDev) [inline packages/echo-core/src/state/useGlobalState.ts:17]: I don't think we have unit tests for this, can you add this? Would be nice to test function and none function

**Review by OveBDev — APPROVED** [review]
Looks good, tested with yalc in echopedia :)

**Comment by CsabaTheKiss** [comment]
> Looks good, tested with yalc in echopedia :)

Thanks, Ove! Will merge it back to the React 18 upgrade branch:
https://github.com/equinor/EchoCore/tree/patch/upgrade-to-react-18

Closing this one, can reference it later.
